IBAnimatable/GradientDesignable.swift

```diff
@@ -78,14 +46,48 @@ fileprivate extension GradientDesignable where Self: UIView {
   }
 
   func configurePredefinedGradient() -> GradientColor? {
-
     guard let gradientType = predefinedGradient else {
       return nil
     }
     return gradientType.colors
   }
 }
 
+public extension GradientDesignable where Self: UINavigationBar {
+  public func configureGradient() {
+    let predefinedGradient = configurePredefinedGradient()
+    if let startColor = startColor, let endColor = endColor {
+      configureGradient(startColor: startColor, endColor: endColor)
+    } else if let startColor = predefinedGradient?.start, let endColor = predefinedGradient?.end {
+      configureGradient(startColor: startColor, endColor: endColor)
+    }
+  }
+}
+
+fileprivate extension GradientDesignable where Self: UINavigationBar {
+
+  func configureGradient(startColor: UIColor, endColor: UIColor) {
+    let gradientLayer = makeGradientLayer()
+    gradientLayer.colors = [startColor.cgColor, endColor.cgColor]
+    gradientLayer.configurePoints(with: startPoint)
+    let image = gradientLayer.image()
+
```

Comment:
Would remove that newline, but add one before the `let image` if you wanted to separate them (up to you)